pkg/core/resources/store/store.go

```diff
@@ -13,6 +13,7 @@ type ResourceStore interface {
 	Create(context.Context, model.Resource, ...CreateOptionsFunc) error
 	Update(context.Context, model.Resource, ...UpdateOptionsFunc) error
 	Delete(context.Context, model.Resource, ...DeleteOptionsFunc) error
+	DeleteMany(context.Context, ...DeleteManyOptionsFunc) error
```

Comment:
I think, `DeleteMany` operation should be implemented at `ResourceManager` level rather than at `ResourceStore` level. Resources should be deleted one-by-one by a `ResourceManager`.

For example, consider an audit feature where we would be required to maintain a log of all operations that modify state of resources. Another example would be complex delete logic per resource type (like in the case of `Mesh`)